packages/line/index.d.ts

```diff
@@ -213,8 +213,17 @@ declare module '@nivo/line' {
     export class Line extends React.Component<LineSvgProps & Dimensions> {}
     export class ResponsiveLine extends React.Component<LineSvgProps> {}
 
+    export interface CustomCanvasLayerProps extends CustomLayerProps {
+        ctx: CanvasRenderingContext2D
+        lineGenerator: any
```

Comment:
Is there a reason you are overriding this to `any`?

Reply:
@wyze I did a bit more testing and added:

```typescript
    export interface LineGenerator {
        (data: Datum[]): string
        x:any
        y:any
        defined:any
        curve:any
        context:any
    }
```

I don't know what the correct typings for `x`, `y`, `defined`, `curve` and `context` are I just know that they where missing.

Reply:
No worries, just pushed a fix. :)